apps/web/src/scenes/NftDetailPage/NftDetailLightbox.tsx

```diff
@@ -0,0 +1,141 @@
+import { useCallback, useEffect, useRef, useState } from 'react';
+import colors from 'shared/theme/colors';
+import styled, { css, keyframes } from 'styled-components';
+
+import { StyledContainer } from '~/contexts/modal/AnimatedModal';
+import CloseIcon from '~/icons/CloseIcon';
+
+type Props = {
+  toggleLightbox: () => void;
+  isLightboxOpen: boolean;
+  tokenId: string;
+};
+
+// This component is used to display an NFT in fullscreen.
+// It is responsible for the transition animations including sizing and fullscreen background.
+// To display the NFT itself, this component simply creates a portal destination in which another component (such as TokenDetailAsset) can place an NFT asset view inside.
+// This is necessary to use the exact same rendered asset when viewing it in on both fullscreen and non-fullscreen modes to achieve a seamless transition.
+export default function NftDetailLightbox({ toggleLightbox, isLightboxOpen, tokenId }: Props) {
+  const ref = useRef<HTMLDivElement>(null);
+  const [distanceFromLeft, setDistanceFromLeft] = useState(0);
+
+  // Because the lightbox is originally positioned absolutely relative to the displayed nft on the detail page, when the lightbox is opened to be fullscreen
+  // we have to manually position it to aligned with the edge of the browser window.
+  useEffect(() => {
+    if (ref.current) {
+      const rect = ref.current.getBoundingClientRect();
+      setDistanceFromLeft(rect.left);
+    }
+  }, []); // Empty dependency array means this runs once on mount
```

Comment:
when the component mounts, `ref` is `null` by default, so there's a chance `if (ref.current)` may not evaluate to doing anything

what we could use is a callbackRef! check this out: https://tkdodo.eu/blog/avoiding-use-effect-with-callback-refs#callback-refs

Reply:
this is elite

Reply:
That's a really cool article!